### PR TITLE
Creates session life tracking cookies and actions

### PR DESCRIPTION
**Pull request by cdmo**
#334 When you have a minute @banukutlu can we talk about how to better organize this new session cookie management? Thinking about making this a separate Module, also thinking about testing.

### CODE REVIEW

File: config/initializers/warden.rb [hunk 1]
```diff
@@ -9,11 +9,14 @@ def authenticate!
...
-            session_token: response['sessionToken'] }
-      success!(u)
+      user = {
+        username: params['user_id'],
+        name: response['name'],
+        patron_key: response['patronKey'],
+        session_token: response['sessionToken']
+      }
+
+      success!(user)
```
Comment (cdmo) [inline config/initializers/warden.rb:19]: this part just came in through a rubocop fix 🤷

**Comment by cdmo** [comment]
This is in QA now

**Review by banukutlu — DISMISSED** [review]
looks good 👍  this branch has been deployed, testing this on qa right now then can be merged

File: spec/features/session_life_spec.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      it 'sets session cookies' do
+        session_began_cookie = page.driver.browser.manage.cookie_named(:session_began)
+        last_active_cookie = page.driver.browser.manage.cookie_named(:last_active)
+
+        expect(session_began_cookie[:value]).to be_present
+        expect(last_active_cookie[:value]).to be_present
+      end
+    end
+
+    xcontext 'when a user has not done anything in the application for a duration greater than the last_active value' do
```
Comment (cdmo) [inline spec/features/session_life_spec.rb:28]: I wasn't able to get a test that properly tests these scenarios, but at least they are documented. The warden test helper for logging in always wants to start a new session with every request.

**Comment by cdmo** [comment]
This is probably ready to go now, we have tested in a remote environment so it's passed that test, probably ready for merging into main

**Review by banukutlu — APPROVED** [review]
👍 looks good

**Pull request merged**